provider/cloudflare/cloudflare.go

```diff
@@ -917,12 +959,15 @@ func groupByNameAndTypeWithCustomHostnames(records DNSRecordsMap, chs CustomHost
 
 		endpoints = append(endpoints, e)
 	}
-
 	return endpoints
 }
 
-// boolPtr is used as a helper function to return a pointer to a boolean
-// Needed because some parameters require a pointer.
-func boolPtr(b bool) *bool {
-	return &b
+// SupportedRecordType returns true if the record type is supported by the provider
+func SupportedRecordType(recordType string) bool {
+	switch recordType {
+	case "MX":
```

Comment:
not covered with unit tests

![Screenshot 2025-06-22 at 10 10 55](https://github.com/user-attachments/assets/6e2b7ceb-9ea7-42c7-bbb7-04336e1ae363)